snuba/state/stateful_count.py

```diff
@@ -0,0 +1,59 @@
+from time import time
+from typing import Sequence, Tuple
+
+from arroyo.processing.strategies.dead_letter_queue import CountInvalidMessagePolicy
+from arroyo.processing.strategies.dead_letter_queue.policies.abstract import (
+    InvalidMessage,
+)
+from arroyo.types import TPayload
+
+from snuba.redis import redis_client
+
+
+class StatefulCountInvalidMessagePolicy(CountInvalidMessagePolicy[TPayload]):
+    """
+    An extension of the CountInvalidMessagePolicy which is able to save and load
+    the state of counted hits in Redis
+    """
+
+    def __init__(self, consumer_group_name: str, limit: int, seconds: int = 60) -> None:
+        self.__name = f"dlq:{consumer_group_name}"
+        self.__seconds = seconds
+        super().__init__(limit, seconds, self._load_state())
+
+    def handle_invalid_message(self, e: InvalidMessage) -> None:
+        """
+        Asynchronously updates Redis hash with a new hit while
```

Comment:
Can you update this comment now the thread got removed?

Reply:
I removed it in general since the method is pretty simple